xbmc/addons/Addon.cpp

```diff
@@ -501,9 +501,9 @@ void OnPreInstall(const AddonPtr& addon)
   addon->OnPreInstall();
 }
 
-void OnPostInstall(const AddonPtr& addon, bool update, bool modal)
+void OnPostInstall(const AddonPtr& addon, bool update, bool modal, int optionalTypeDependend)
```

Comment:
maybe some enum class can be used here instead?

Reply:
As enum not the best for there, as it can come thats an other add-on type can use this in future too, but there e.g. as integer. That's make it hard to have usable in all possible conditions.

Reply:
I'd consider reframing the problem with an enum, because it took me a while to track down `POST_INSTALL_LANGUAGE_RESOURCE_NO_SELECT` as the only non-negative option. What does zero mean?

Reply:
Changed to use an `enum` for it.